src/sema/expression/function_call.rs

```diff
@@ -2620,3 +2546,81 @@ pub(crate) fn function_type(func: &Function, external: bool, resolve_to: Resolve
         }
     }
 }
+
+/// This function evaluates the arguments of a function call with either positional arguments or
+/// named arguments.
+fn evaluate_argument(
+    arg: &pt::Expression,
+    context: &ExprContext,
+    ns: &mut Namespace,
+    symtable: &mut Symtable,
+    arg_ty: &Type,
+    errors: &mut Diagnostics,
+    cast_args: &mut Vec<Expression>,
+) -> bool {
+    let arg = match expression(arg, context, ns, symtable, errors, ResolveTo::Type(arg_ty)) {
+        Ok(e) => e,
+        Err(_) => {
+            return false;
+        }
+    };
+
+    match arg.cast(&arg.loc(), arg_ty, true, ns, errors) {
+        Ok(expr) => cast_args.push(expr),
+        Err(_) => return false,
+    }
+
+    true
+}
+
+/// This function finishes resolving internal function calls
+fn resolve_internal_call(
+    loc: &Loc,
+    function_no: usize,
+    context: &ExprContext,
+    resolve_to: ResolveTo,
+    virtual_call: bool,
+    cast_args: Vec<Expression>,
+    ns: &Namespace,
+    errors: &mut Diagnostics,
+) -> Option<Expression> {
+    let func = &ns.functions[function_no];
+
+    if func.contract_no != context.contract_no && func.is_private() {
+        errors.push(Diagnostic::error_with_note(
+            *loc,
+            format!("cannot call private {}", func.ty),
+            func.loc,
+            format!("declaration of {} '{}'", func.ty, func.name),
+        ));
+
+        return None;
+    } else if func.contract_no == context.contract_no
```

Comment:
inherited `external` functions are callable?

Reply:
Needs test

Reply:
> inherited `external` functions are callable?

Oh, I missed this case.